pkg/controller/statusmanager/pod_status.go

```diff
@@ -84,53 +84,92 @@ func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.No
 		*conditions = append(*conditions, newCondition)
 		return true
 	}
-
-	if existingCondition.Status != newCondition.Status {
+	if (existingCondition.Status != newCondition.Status || existingCondition.Reason != newCondition.Reason ||
+		existingCondition.Message != newCondition.Message) {
+		changed = true
 		existingCondition.Status = newCondition.Status
+		existingCondition.Reason = newCondition.Reason
+		existingCondition.Message = newCondition.Message
 		existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
-		changed = true
 	}
-
-	existingCondition.Reason = newCondition.Reason
-	existingCondition.Message = newCondition.Message
 	return changed
 }
 
-func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, reason string, message string, sharedInfo *sharedinfo.SharedInfo) {
+// CheckExistingAgentPods is for a case: nsx-node-agent becomes unhealthy -> NetworkUnavailable=True -> operator off ->
+// nsx-node-agent becomes healthy and keeps running -> operator up -> operator cannot receive nsx-node-agent event to
+// set NetworkUnavailable=False. So a full sync at the start time is necessary.
+func (status *StatusManager) CheckExistingAgentPods (firstBoot *bool, sharedInfo *sharedinfo.SharedInfo) {
+	status.Lock()
+	defer status.Unlock()
+
+	if *firstBoot {
+		*firstBoot = false
+	} else {
+		return
+	}
+	log.Info("Checking all nsx-node-agent pods for node condition")
+	pods := &corev1.PodList{}
+	err := status.client.List(context.TODO(), pods, client.MatchingLabels{"component": operatortypes.NsxNodeAgentContainerName})
+	if err != nil {
+		log.Error(err, "Error getting pods for node condition")
```

Comment:
Do we need to set firstBoot to false here and retry?

Reply:
make sense & done